Core/src/org/sleuthkit/autopsy/report/TableReportGenerator.java

```diff
@@ -152,52 +156,81 @@ private void makeBlackboardArtifactTables() {
                 continue;
             }
 
-            /*
-                 Gets all of the attribute types of this artifact type by adding
-                 all of the types to a set
-             */
-            Set<BlackboardAttribute.Type> attrTypeSet = new TreeSet<>((BlackboardAttribute.Type o1, BlackboardAttribute.Type o2) -> o1.getDisplayName().compareTo(o2.getDisplayName()));
-            for (ArtifactData data : artifactList) {
-                List<BlackboardAttribute> attributes = data.getAttributes();
-                for (BlackboardAttribute attribute : attributes) {
-                    attrTypeSet.add(attribute.getAttributeType());
+            /* TSK_ACCOUNT artifacts get grouped by their TSK_ACCOUNT_TYPE
+             * attribute, and then handed off the default method for writing
+             * tables. */
+            if (type.getTypeID() == BlackboardArtifact.ARTIFACT_TYPE.TSK_ACCOUNT.getTypeID()) {
```

Comment:
I think we need some additional documentation for this cleverness. Let's say something about the way the synthesized data type is of the form  [BlackboardArtifact.ARTIFACT_TYPE.TSK_ACCOUNT.displayName()] :  [BlackboardAttribute.ATTRIBUTE_TYPE.TSK_ACCOUNT.getValueString].